### ISSUE TITLE
Fail to start build

### ISSUE THREAD

**Issue by hanjh1986** [ansible/awx#55]
I installed all Prerequisites, and then try to your follow installation

Ansible
gettext package for your platform (See Installing gettext)
Docker
docker-py Python module
Node 6.x LTS version
NPM 3.x LTS

...
Start the build

If you are not pushing images to a Docker registry, start the build by running the following:

# Set the working director to installer
$ cd installer

# Run the Ansible playbook
**$ ansible-playbook -i inventory install.yml**
...

after run the ansible playbook, 
I face this screen(+with error message)
It just seems like SSL error or something relative

when I start docker only,
docker makes images, containers good.

...
...

TASK [image_build : Clean distribution] ***************************************************************************************************************************************
changed: [localhost -> localhost]

TASK [image_build : Build AWX distribution] ***********************************************************************************************************************************

fatal: [localhost -> localhost]: FAILED! => {"changed": true, "cmd": "make sdist", "delta": "0:05:24.711488", "end": "2017-09-08 15:46:01.046463", "failed": true, "rc": 2, "start": "2017-09-08 15:40:36.334975", "stderr": "npm ERR! fetch failed https://registry.npmjs.org/abbrev/-/abbrev-1.1.0.tgz\nnpm WARN retry will retry, error on last attempt: Error: tunneling socket could not be established, cause=write EPROTO 140373632497472:error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol:../deps/openssl/openssl/ssl/s23_clnt.c:797:\nnpm WARN retry \nnpm ERR! fetch failed https://registry.npmjs.org/xmlhttprequest-ssl/-/xmlhttprequest-ssl-1.5.3.tgz\nnpm WARN retry will retry, error on last attempt: Error: tunneling socket could not be established, cause=write EPROTO 140373632497472:error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol:../deps/openssl/openssl/ssl/s23_clnt.c:797:\nnpm WARN retry \nnpm ERR! fetch failed https://registry.npmjs.org/xmlbuilder/-/xmlbuilder-8.2.2.tgz\nnpm WARN retry will retry, error on last attempt: Error: tunneling socket could not be established, cause=write EPROTO 140373632497472:error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol:../deps/openssl/openssl/ssl/s23_clnt.c:797:\nnpm WARN retry \nnpm ERR! fetch failed https://registry.npmjs.org/wtf-8/-/wtf-8-1.0.0.tgz\nnpm WARN retry will retry, error on last attempt: Error: tunneling socket could not be established, cause=write EPROTO 140373632497472:error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol:../deps/openssl/openssl/ssl/s23_clnt.c:797:\nnpm WARN retry \nnpm ERR! fetch failed https://registry.npmjs.org/ws/-/ws-1.1.1.tgz\nnpm WARN retry will retry, error on last attempt: Error: tunneling socket could not be established, cause=write EPROTO 140373632497472:error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol:../deps/openssl/openssl/ssl/s23_clnt.c:797:\nnpm WARN retry \nnpm ERR! fetch failed https://registry.npmjs.org/write/-/write-0.2.1.tgz\nnpm WARN retry will retry, error on last attempt: Error: tunneling socket could not be established, cause=write EPROTO 140373632497472:error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol:../deps/openssl/openssl/ssl/s23_clnt.c:797:\nnpm WARN retry ........

please, give me how to solve this problem...

**Comment by uriinf** [comment]
Same for Fedora26 and Ubuntu
$ node -v
v6.11.2
$ npm -v
3.10.10

**Comment by hanjh1986** [comment]
I found something. 
This error like this

.....git://github.com/ansible/angular-codemirror....
...git://github.com/ansible/angular-drag-and-drop-lists.git...

git?? I tried this url to my chrome and fail.
but, chang git to https, success

git://github.com/ansible/angular-drag-and-drop-lists.git -> https://github.com/ansible/angular-drag-and-drop-lists.git

maybe there *.git has dependency, but, that, git://github.com/ansi~, url is wrong I think

**Comment by jains99** [comment]
I am also facing issues in same step,but in my case errors are different. I am running it as root user.

`TASK [image_build : Build AWX distribution] **********************************************************************************************************
fatal: [localhost -> localhost]: FAILED! => {"changed": true, "cmd": "make sdist", "delta": "0:00:12.254391", "end": "2017-09-08 15:53:47.653031", "failed": true, "rc": 2, "start": "2017-09-08 15:53:35.398640", "stderr": "npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but npm-shrinkwrap.json was generated for lockfileVersion@0. I'll try to do my best with it!\nnpm ERR! Error while executing:\nnpm ERR! /bin/git ls-remote -h -t git://github.com/ansible/angular-breadcrumb.git\nnpm ERR! \nnpm ERR! fatal: Could not change back to '/root/awx': Permission denied\nnpm ERR! \nnpm ERR! exited with error code: 128\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /root/.npm/_logs/2017-09-08T10_23_47_598Z-debug.log\nmake: *** [awx/ui/.deps_built] Error 1", "stderr_lines": ["npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but npm-shrinkwrap.json was generated for lockfileVersion@0. I'll try to do my best with it!", "npm ERR! Error while executing:", "npm ERR! /bin/git ls-remote -h -t git://github.com/ansible/angular-breadcrumb.git", "npm ERR! ", "npm ERR! fatal: Could not change back to '/root/awx': Permission denied", "npm ERR! ", "npm ERR! exited with error code: 128", "", "npm ERR! A complete log of this run can be found in:", "npm ERR!     /root/.npm/_logs/2017-09-08T10_23_47_598Z-debug.log", "make: *** [awx/ui/.deps_built] Error 1"], "stdout": "npm --unsafe-perm --prefix awx/ui install awx/ui", "stdout_lines": ["npm --unsafe-perm --prefix awx/ui install awx/ui"]}`

**Comment by szibis** [comment]
I have this error

```
        "",
        "Traceback (most recent call last):",
        "  File \"tools/scripts/compilemessages.py\", line 154, in <module>",
        "    output, errors, status = popen_wrapper(args)",
        "  File \"tools/scripts/compilemessages.py\", line 44, in popen_wrapper",
        "    (args[0], strerror)), sys.exc_info()[2])",
        "Exception: (<type 'exceptions.StandardError'>, StandardError('Error executing msgfmt: No such file or directory',), <traceback object at 0x101a6b680>)",
        "make: *** [.i18n_built] Error 1"
```

**Comment by fsidnenko** [comment]
@jains99 
It seems that the version of npm is not 3.x
Try to do so: npm install npm@3.0 -g

**Comment by jains99** [comment]
@fsidnenko yes it was 5.4.1 , later I tried with 3.0 but getting below error -

`TASK [image_build : Build AWX distribution] **********************************************************************************************************
fatal: [localhost -> localhost]: FAILED! => {"changed": true, "cmd": "make sdist", "delta": "0:00:48.794235", "end": "2017-09-08 18:26:21.591053", "failed": true, "rc": 2, "start": "2017-09-08 18:25:32.796818", "stderr": "(node:3014) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.\nnpm ERR! git clone --template=/root/.npm/_git-remotes/_templates --mirror git://github.com/ansible/angular-tz-extensions.git /root/.npm/_git-remotes/git-github-com-ansible-angular-tz-extensions-git-33caaa9ccf5dfe29a95962c17c3c9e6b9775be35-302f09e35af3690ce351391cd3028601: Cloning into bare repository '/root/.npm/_git-remotes/git-github-com-ansible-angular-tz-extensions-git-33caaa9ccf5dfe29a95962c17c3c9e6b9775be35-302f09e35af3690ce351391cd3028601'...\nnpm ERR! git clone --template=/root/.npm/_git-remotes/_templates --mirror git://github.com/ansible/angular-tz-extensions.git /root/.npm/_git-remotes/git-github-com-ansible-angular-tz-extensions-git-33caaa9ccf5dfe29a95962c17c3c9e6b9775be35-302f09e35af3690ce351391cd3028601: \nnpm ERR! git clone --template=/root/.npm/_git-remotes/_templates --mirror git://github.com/ansible/angular-tz-extensions.git /root/.npm/_git-remotes/git-github-com-ansible-angular-tz-extensions-git-33caaa9ccf5dfe29a95962c17c3c9e6b9775be35-302f09e35af3690ce351391cd3028601: fatal: unable to connect to github.com:\nnpm ERR! git clone --template=/root/.npm/_git-remotes/_templates --mirror git://github.com/ansible/angular-tz-extensions.git /root/.npm/_git-remotes/git-github-com-ansibl.......`

**Comment by ghost** [comment]
Some things I've found:

- For CentOS 7 minimal install you need to install bzip2
- If you get `contextify@0.1.15 install: node-gyp rebuild` error during `image_build : Build AWX distribution`, remove $HOME/.node-gyp

**Comment by gconsidine** [comment]
For issues fetching npm packages via GitHub, you might try setting this configuration option as a short-term solution: `git config --global url."https://".insteadOf git://`

I'm going to experiment with changing dependencies in `package.json` from this format:

`"angular-breadcrumb": "github:ansible/angular-breadcrumb#0.4.1"`

to:

`"angular-breadcrumb":"git+https://git@github.com:ansible/angular-breadcrumb.git#0.4.1"`

I'll report back in a bit.

**Comment by gconsidine** [comment]
Can anyone verify #68 as a fix for this issue? Thanks!

As a note, the correct versions of npm and Node.js to build the UI are:

* Node.js 6.x LTS (currently v6.11.3)
* npm 3.x (currently v3.10.10 -- comes bundled with Node.js v6.11.3)

**Comment by gconsidine** [comment]
Fixed in #49
